### PR TITLE
added test that needs to "undo" a checkpoint. currently, it fails.

### PR DESCRIPTION
**Pull request by dahliamalkhi**
this seems to be a bug with checkpoint/recovery. it fails both on master and on #669

### CODE REVIEW

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **be77e86**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #670 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/Lr6Mapg3I43ZTJ8v4npsJrq07e3EcHwA)

**Comment by no2chem** [comment]
@dahliamalkhi Okay, this test will still fail on #669, but that's because it doesn't catch the trimmed transaction abort. Could you modify it to catch that?

Also, while you're at it, can you move the tests from CheckpointSmokeTest to CheckpointTest? I'm not really sure what the difference is, and "Smoke" isn't very informative (are they supposed to check if the checkpoints will catch on fire?).

**Comment by dahliamalkhi** [comment]
yes, I will be happy to do it

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **2a0293a**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #670 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/aQCl2Y2JsO4kzvgkJ4022OWYuBATglBN)

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **a9ed5d9**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #670 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/4EP3ohrrBJcCxbeC3EeXCMhIbXxmoSQj)

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java [hunk 1]
```diff
@@ -368,6 +368,22 @@ public void periodicCkpointTrimTest() throws Exception {
         }
     }
 
+    /** We have some non-determinism that needs more runtime to
+     *  try to find buggy executions.  These tests (and the wall
+     *  clock time they consume on each test run) can go away
+     *  sometime, after "soak time" and another round of review.
+     */
+    @Test public void periodicCkpointTrimTest0() throws Exception { periodicCkpointTrimTest(); }
```
Comment (Maithem) [inline test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java]: Hmm, just use a loop?

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **b2e1143**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #670 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/J2S3xvbc8fsz7aTWZy0JeDAPaRnqAkm3)

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java (hunk 1 above)
Comment (slfritchie) [inline test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java]: @Maithem The simplest loop would hide how frequently the test fails. This way gives more verbose reporting. {shrug} This is intended to be temporary, until I finish alternative test technique to force different thread execution schedules.

**Comment by slfritchie** [comment]
After discussion with @dahliamalkhi and @no2chem, we'll close this and re-open with smaller PRs.  This PR is too tangled with #676's changes to keep as-is.

**Pull request closed without merging**